core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -854,199 +864,156 @@ public Graph build() {
-        }
+        private StoreProperties applyParentStoreProperties(final StoreProperties properties, final StoreProperties parentStoreProperties) {
+            // Start with the parent store properties
+            StoreProperties mergedProperties = parentStoreProperties;
 
-        private boolean hasHook(final List<GraphHook> hooks, final Class<? extends GraphHook> hookClass) {
-            for (final GraphHook hook : hooks) {
-                if (hookClass.isAssignableFrom(hook.getClass())) {
-                    return true;
-                }
+            // Apply rest of properties
+            if (properties != null) {
+                mergedProperties.merge(properties);
             }
-            return false;
+            return mergedProperties;
         }
 
-        private void updateSchema(final GraphConfig config) {
-            Schema mergedParentSchema = null;
-
-            if (null != parentSchemaIds) {
-                for (final String parentSchemaId : parentSchemaIds) {
-                    if (null != parentSchemaId) {
-                        final Schema parentSchema = config.getLibrary().getSchema(parentSchemaId);
-                        if (null != parentSchema) {
-                            if (null == mergedParentSchema) {
-                                mergedParentSchema = parentSchema;
-                            } else {
-                                mergedParentSchema = new Schema.Builder()
-                                        .merge(mergedParentSchema)
-                                        .merge(parentSchema)
-                                        .build();
-                            }
-                        }
-                    }
-                }
-            }
-
-            if (null != mergedParentSchema) {
-                if (null == schema) {
-                    schema = mergedParentSchema;
+        /**
+         * Merges the supplied schema with any parent schemas and returns the
+         * result.
+         *
+         * @param currentSchema The current schema.
+         * @param parentSchemas The list of parent schemas to merge.
+         * @return Merged schema of any parents and the supplied schema.
+         */
+        private Schema applyParentSchemas(final Schema currentSchema, final List<Schema> parentSchemas) {
+            Schema mergedSchema = null;
+
+            // Apply parent schemas
+            for (final Schema parentSchema : parentSchemas) {
+                // If the merged result is still null init with the current schema else merge
+                if (mergedSchema == null) {
+                    mergedSchema = parentSchema;
                 } else {
-                    schema = new Schema.Builder()
-                            .merge(mergedParentSchema)
-                            .merge(schema)
-                            .build();
+                    mergedSchema = new Schema.Builder()
+                        .merge(mergedSchema)
```

Comment:
I might be wrong but I think the logic is different as if `addSchema()` was used then the order that the shemas are merged may be wrong e.g. currently the parent schemas will always be added first then any existing ones merged later. If I used `addSchema()` the parent ones could be merged after, this might not be an issue but I'm unsure.